packages/console/src/components/CreateTenantModal/SelectTenantPlanModal/PlanCardItem/index.tsx

```diff
@@ -0,0 +1,120 @@
+import { maxFreeTenantLimit } from '@logto/schemas';
+import classNames from 'classnames';
+import { useContext } from 'react';
+import { Trans, useTranslation } from 'react-i18next';
+
+import ArrowRight from '@/assets/icons/arrow-right.svg';
+import PlanDescription from '@/components/PlanDescription';
+import PlanName from '@/components/PlanName';
+import PlanQuotaList from '@/components/PlanQuotaList';
+import { ReservedPlanId } from '@/consts/subscriptions';
+import { TenantsContext } from '@/contexts/TenantsProvider';
+import Button from '@/ds-components/Button';
+import DangerousRaw from '@/ds-components/DangerousRaw';
+import DynamicT from '@/ds-components/DynamicT';
+import TextLink from '@/ds-components/TextLink';
+import { type SubscriptionPlanQuota, type SubscriptionPlan } from '@/types/subscriptions';
+
+import * as styles from './index.module.scss';
+
+const featuredQuotaKeys: Array<keyof SubscriptionPlanQuota> = [
+  'mauLimit',
+  'machineToMachineLimit',
+  'standardConnectorsLimit',
+  'rolesLimit',
+  'scopesPerRoleLimit',
+  'auditLogsRetentionDays',
+];
+
+type Props = {
+  plan: SubscriptionPlan;
+  onSelect: () => Promise<void>;
+};
+
+function PlanCardItem({ plan, onSelect }: Props) {
+  const { t } = useTranslation(undefined, { keyPrefix: 'admin_console.upsell.create_tenant' });
+  const { tenants } = useContext(TenantsContext);
+  const { stripeProducts, id: planId, name: planName, quota } = plan;
+
+  const basePrice =
+    stripeProducts.find(({ type }) => type === 'flat')?.price.unitAmountDecimal ?? 0;
+
+  const tierPrices = stripeProducts
+    .filter(({ type }) => type !== 'flat')
+    .map(({ price }) => {
+      return price.unitAmountDecimal;
+    });
+
+  const isFreePlan = planId === ReservedPlanId.free;
+
+  // Todo: @xiaoyijun filter our all free tenants
+  const isFreeTenantExceeded = tenants.length >= maxFreeTenantLimit;
+
+  return (
+    <div className={styles.container}>
+      <div className={styles.planInfo}>
+        <div className={styles.title}>
+          <PlanName name={planName} />
+        </div>
+        <div className={styles.priceInfo}>
+          <div className={styles.priceLabel}>{t('base_price')}</div>
+          <div className={styles.price}>
+            ${t('monthly_price', { value: Number(basePrice) / 100 })}
+          </div>
+          <div className={styles.priceLabel}>
+            {t('mau_unit_price')}
+            <span className={styles.unitPrices}>
+              {tierPrices.length > 0
+                ? tierPrices.map((price) => `$${Number(price) / 100}`).join(' ')
+                : '$0.00'}
```

Comment:
I think this can be calculated in line 42 as well. And don't forget to `useMemo`